### PR TITLE
Zbig01 backup update 1

### PR DESCRIPTION
**Pull request by bradymiller**
User interface improvements to the Administration->Backup along with a help script. Taken from the following PR:
https://github.com/openemr/openemr/pull/1235

Also fixes to the commandline restore script:
"The first thing that I noted was that whatever your database and openemr installation was called the recovered one would be called openemr. Secondly the restore script assumed the Web root directory to be /var/www/
I order to fix these problems I created a text file that contains the installation data, openemr-setup.txt, and modified the restore script to accept these values as needed. This now results in the recovered installation to be identical in all respects to the original. In addition the restore script is now included in the emr_backup.tar which now contains 2 archives and two files - openemr-setup.txt and restore."

Demo is here:
http://www.open-emr.org/wiki/index.php/Development_Demo#Epsilon_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by bradymiller** [comment]
Up for grabs demo is here:
http://www.open-emr.org/wiki/index.php/Development_Demo#Epsilon_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
@zbig01 ,
While the openemr-setup.txt is helpful, my initial impression is that including the database password is going to be a security vulnerability. Let do some more analysis of the code though.
-brady

**Comment by bradymiller** [comment]
Disregard my post above. Not an issue since these credentials are also in sqlconf.php script which is in the backup.

File: contrib/util/restore [hunk 1]
```diff
@@ -163,16 +182,16 @@ while [ $LOOKING -eq 1 ]; do
 done
 
 # Extract the OpenEMR web directory tree.
-dlg_info "Extracting $BAKDIR/openemr.tar.gz ..."
-mkdir openemr
-cd openemr
-tar zxf ../openemr.tar.gz
+dlg_info "Extracting $BAKDIR/$webroot_name.tar.gz ..."
+mkdir $webroot_name
```
Comment (sunsetsystems) [inline contrib/util/restore]: I think "openemr" is a temporary directory here, so shoudn't matter what its name is?

File: contrib/util/restore [hunk 2]
```diff
@@ -154,6 +158,21 @@ while [ $LOOKING -eq 1 ]; do
     dlg_info "Extracting $TARFILE ..."
     cd $BAKDIR
     tar -xf $TARFILE
+    #RP_ADDED to get names of OEDIR ans webroot name if different from default by reading values from openemr_setup.txt
+    filename=$BAKDIR/openemr_setup.txt
```
Comment (sunsetsystems) [inline contrib/util/restore]: I think you should allow for the case where openemr_setup.txt does not exist, which will be the case for old backups and perhaps also backups created with a home-grown method.

**Comment by sunsetsystems** [comment]
Note the restore script does already give you the option of specifying the target directory, database name, password etc. Also make sure it still works right when restoring multiple sites to the same target directory.

I wonder about the desirability of translating entire web pages via the current translation system. But I guess that's a matter for another day.

**Comment by bradymiller** [comment]
hi @zbig01 ,
Check out above comments by myself and @sunsetsystems . Regarding translations of the help page, my opinion is that this is necessary if we include instructions like you are doing (ie. would keep it). I'm still in process of testing/reviewing this.
-brady

**Comment by robertdown** [comment]
Closing this since outdated PR

**Pull request closed without merging**